api/spec/src/types.tsp

```diff
@@ -109,39 +132,54 @@ model Resource {
  * The unit of time for the interval.
  * One of: `day`, `week`, `month`, or `year`.
  */
-@friendlyName("IntervalUnit")
-union IntervalUnit {
-  @summary("Day")
-  day: "day",
-
-  @summary("Week")
-  week: "week",
-
-  @summary("Month")
-  month: "month",
-
-  @summary("Year")
-  year: "year",
+@friendlyName("RecurringPeriodEnum")
+enum Interval {
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  Day: "DAY",
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  Week: "WEEK",
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  Month: "MONTH",
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  Year: "YEAR",
 }
 
 /**
- * Time interval with a time unit and count.
+ * Recurring period with an interval and an anchor
  */
-@example(#{ unit: "month", count: 3 })
-@friendlyName("Interval")
-model Interval {
+@example(#{ interval: Interval.Day, anchor: DateTime.fromISO("2023-01-01T01:01:01.001Z") })
+@friendlyName("RecurringPeriod")
+model RecurringPeriod {
```

Comment:
Same. Also, we'll use `duration` (ISO 8601) in place of current `Interval` object.